docs/decisions/broker-trigger.md

```diff
@@ -0,0 +1,137 @@
+is successful, future recommendations for Source CRD authors would be to default
+the Source's `spec.sink` attribute to point to the `default` Broker in the
+namespace.
+
+Historical storage and replay of events is not part of the Broker design or MVP.
+
+Broker implements the "Addressable" duck type to allow Source custom resources
+to target the Broker. Broker DNS names should also be predictable (e.g.
+`default-broker` for the `default` Broker), to enable whitebox event producers
+to target the Broker without needing to reconcile against the "Addressable" duck
+type.
+
+#### Broker Object definition
+
+```golang
+type Broker struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec BrokerSpec `json:"spec,omitempty"`
+	Status BrokerStatus `json:"status,omitempty"`
+}
+
+type BrokerSpec struct {
+  // Empty for now
+}
+
+type BrokerStatus struct {
+	Conditions duckv1alpha1.Conditions `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type"`
+
+	Address duckv1alpha1.Addressable `json:"address,omitempty"`
+}
+```
+
+### Trigger
+
+A Trigger represents a registration of interest (request for delivery to an
+Addressable object or URL) in a filtered selection of events delivered to a
+Broker. For the MVP, Triggers may only target Addressable objects in the same
+namespace.
+
+#### Buffering
+
+Triggers are assumed to buffer (durably queue) filtered events from the Broker
+from the time they are created, and to deliver the buffered events to the target
+Addressable as it is able to receive them.
+
+#### Filtering
+
+An event consumer which is interested in several sets of events may need to
+create multiple Triggers, each with an ANDed set of filter conditions which
+select the events. In the case of multiple Triggers with the same target whose
+filter conditions select an overlapping set of events, events will be delivered
+once for each Trigger which selects the event (i.e. Trigger selection criteria
+are independent, not aggregated in the Broker).
+
+At some later point, we expect that a more complex set of filtering expressions
+will be supported. For the initial MVP, exact match on the CloudEvents
+[`type`](https://github.com/cloudevents/spec/blob/master/spec.md#type) and
+[`source`](https://github.com/cloudevents/spec/blob/master/spec.md#source)
+attributes will be supported.
+
+#### Trigger Object Definition
+
+```golang
+type Trigger struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec TriggerSpec `json:"spec,omitempty"`
+	Status TriggerStatus `json:"status,omitempty"`
+}
+
+type TriggerSpec struct {
+	// Defaults to 'default'.
+	Broker string `json:"broker,omitempty"`
+  Filter *TriggerFilter `json:"filter,omitempty"`
+  // As from Subscription's `spec.subscriber`
+	Target *SubscriberSpec `json:"target,omitempty"`
+}
+
+type TriggerFilter struct {
+  // Exact match expressions; if empty, all values are matched.
```

Comment:
indent off here.

Reply:
Also wording is a bit funny with L85 that says exact match on type / source is supported.

Reply:
Updated L85 to say that "all match or exact match" are allowed. (I thought it was implied that the fields could be left out entirely for an all match, but it's good to be explicit.)